platform-hub-api/app/controllers/kubernetes/tokens_controller.rb

```diff
@@ -0,0 +1,75 @@
+class Kubernetes::TokensController < ApiJsonController
+
+  before_action :find_identity
+
+  # GET /:user_id
+  def index
+    authorize! :read, @identity
+    render json: Kubernetes::TokenService.tokens_from_identity(@identity)
+  end
+
+  # PATCH/PUT /:user_id/:cluster
+  def create_or_update
+    authorize! :manage, @identity
+    tokens, created_or_updated_token = Kubernetes::TokenService.create_or_update_token(
+      @identity,
+      params[:cluster],
+      params[:token][:groups]
+    )
+
+    unless created_or_updated_token.valid?
+      render_model_errors created_or_updated_token.errors and return
+    end
+
+    @identity.data[:tokens] = tokens
+
+    if @identity.save
+      AuditService.log(
+        context: audit_context,
+        action: 'update_kubernetes_identity',
+        auditable: @identity,
+        comment: "Kubernetes `#{params[:cluster]}` token created or updated for user '#{@identity.user.email}' - Assigned groups: #{created_or_updated_token.groups}"
+      )
+
+      render json: created_or_updated_token
+    else 
+      render_model_errors @identity.errors
+    end
+  end
+
+  # DELETE /:user_id/:cluster
+  def destroy
+    authorize! :manage, @identity
+    tokens, _ = Kubernetes::TokenService.delete_token(@identity, params[:cluster])
```

Comment:
New kubernetes token requires identity as it's associated with it but can change that to take a token array instead and identity id when needed.